contracts/StablecoinConverter.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.5.0;
+
+import "./EpochTokenLocker.sol";
+import "./libraries/IdToAddressBiMap.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "./libraries/IdToAddressBiMap.sol";
+
+
+contract StablecoinConverter is EpochTokenLocker {
+    using SafeMath for uint;
+    using BytesLib for bytes32;
+
+    event OrderPlacement(
+        address owner,
+        uint16 buyToken,
+        uint16 sellToken,
+        bool isSellOrder,
+        uint32 validFrom,
+        uint32 validUntil,
+        uint256 buyAmount,
+        uint256 sellAmount
+    );
+
+    event OrderCancelation(
+        address owner,
+        uint id
+    );
+
+    //outstanding volume of an order is encoded as buyAmount or sellAmount depending on isSellOrder
+    struct Order {
+        uint16 buyToken;
+        uint16 sellToken;
+        uint32 validFrom;  // order is valid from auction collection period: validFrom inclusively
+        uint32 validUntil;  // order is valid till auction collection period: validUntil inclusively
+        bool isSellOrder;
+        uint128 buyAmount;
+        uint128 sellAmount;
+    }
+
+    // User-> Order
+    mapping(address => Order[]) public orders;
+
+    IdToAddressBiMap.Data private registeredTokens;
+    uint public MAX_TOKENS;
+    uint16 public numTokens = 0;
+
+    constructor(uint maxTokens) public {
+        MAX_TOKENS = maxTokens;
+    }
+
+    function addToken(address _tokenAddress) public {
+        require(numTokens < MAX_TOKENS, "Max tokens reached");
+        require(
+            IdToAddressBiMap.insert(registeredTokens, numTokens, _tokenAddress),
+            "Token already registered"
+        );
+        numTokens++;
+    }
+
+    function placeOrder(
+        uint16 buyToken,
+        uint16 sellToken,
+        bool isSellOrder,
+        uint32 validUntil,
+        uint128 buyAmount,
+        uint128 sellAmount
+    ) public returns (uint) {
+        orders[msg.sender].push(Order({
+            buyToken: buyToken,
```

Comment:
does truffle give any warnings about reusing an internal variable name for a function parameter?

Reply:
nope :)